subsys/net/ip/net_context.c

```diff
@@ -1573,11 +1573,18 @@ static struct net_pkt *context_alloc_pkt(struct net_context *context,
 {
 	struct net_pkt *pkt;
 
+	/* We reference it once again to make sure context is not going
+	 * to disappear while allocating a packet
```

Comment:
What would make it disappear?

Reply:
is there any guaranty that net_context_unref() is not going to be called one time too many (or net_context_put) in the mean time? allocation might take a while depending on load. The application could close the socket (thus the context) for instance.

Reply:
> The application could close the socket (thus the context) for instance.

Right. And as I mentioned in the next comment (in the main comment thread), this touches on other (proposed) changes, e.g. https://github.com/zephyrproject-rtos/zephyr/pull/9565 , which vice-versa try to remove ref/unref wrapping, about which I came to conclusion that it's superfluous. Is it right or not? I wish we had a clear rules and understanding where we need to lock/unlock, and where ref/unref. As certainly, a case addressed by this PR is not the only one.